var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -241,3 +244,55 @@ def cmake_args(self):
             args.append('-DCUDA_ARCH=' + self.get_cuda_arch_string(cuda_arch))
 
         return args
+
+    # TODO: Replace this method and its 'get' use for cmake path with
+    #   join_path(self.spec['cmake'].prefix.bin, 'cmake') once stand-alone
+    #   tests can access build dependencies through self.spec['cmake'].
+    def cmake_bin(self, set=True):
+        """(Hack) Set/get cmake dependency path."""
+        filepath = join_path(self.install_test_root, 'cmake_bin_path.txt')
+        if set:
+            with open(filepath, 'w') as out_file:
+                cmake_bin = join_path(self.spec['cmake'].prefix.bin, 'cmake')
+                out_file.write('{0}\n'.format(cmake_bin))
+        else:
+            with open(filepath, 'r') as in_file:
+                return in_file.read().strip()
+
+    @run_after('build')
+    def setup_smoke_test(self):
+        """Skip setup smoke tests for AMReX versions less than 21.12."""
+        if self.spec.satisfies('@:21.11'):
+            return
+
+        self.cache_extra_test_sources(['Tests'])
+
+        # TODO: Remove once self.spec['cmake'] is available here
+        self.cmake_bin(set=True)
+
+    def test(self):
+        """Skip smoke tests for AMReX versions less than 21.12."""
+        if self.spec.satisfies('@:21.11'):
+            print("SKIPPED: Stand-alone tests not supported for this version of AMReX.")
+            return
+
+        """Perform smoke tests on installed package."""
+        # TODO: Remove/replace once self.spec['cmake'] is available here
+        cmake_bin = self.cmake_bin(set=False)
+
+        args = []
+        args.append('-S ./cache/amrex/Tests/SpackSmokeTest')
+        args.append('-DAMReX_ROOT=' + self.prefix)
+        args.extend(self.cmake_args())
```

Comment:
The following needs to be defined to ensure the correct MPI is being used to build the tests:

```suggestion
        args.append('-DAMReX_ROOT=' + self.prefix)
        args.append('-DMPI_C_COMPILER=' + self.spec['mpi'].mpicc)
        args.append('-DMPI_CXX_COMPILER=' + self.spec['mpi'].mpicxx)
        args.extend(self.cmake_args())
```

Reply:
Possible, or we just hint again `MPI_ROOT` (or CMAKE_PREFIX_PATH) via the environment.
I didn't want to suggest that route, because technically a robust solution should hint _all_ dependencies again, since `find_dependency()` might need them when we pick up the AMReX target.

Reply:
Defining `MPI_ROOT` to `self.spec['mpi'].prefix` also works.

Not sure I understand the `mix-and-match` comment.  The concretized `mpi` dependency should, in theory, resolve appropriately during smoke testing.

Reply:
I think I was mistaken, `MPI_<lang>_COMPILER` is not only a cached option but also a hint for `FindMPI.cmake`: https://cmake.org/cmake/help/latest/module/FindMPI.html#variables-for-locating-mpi